### PR TITLE
add premerge ci

### PR DESCRIPTION
**Pull request by stuarthu**
### SUMMARY
Add premerge ci for superset repo

### TEST PLAN
All successive PR including this one will trigger superset-premerge pipeline and run.
PR can be merged only if premerge ci is success.

### CODE REVIEW

File: .buildkite/premerge.steps.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+common: &common
+  agents:
+    - "agent_count=8"
+    - "boot_disk_size_gb=400"
+    - "capable_of_building=platform"
+    - "machine_type=quad"
```
Comment (Helcaraxan) [inline .buildkite/premerge.steps.yaml:7]: This is way too big (and costly) a machine for this pipeline, better to use a `half` with `agent_count=1`.

File: .buildkite/premerge.steps.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - "queue=${CI_LINUX_BUILDER_QUEUE:-v4-20-06-23-173521-bk12705-2627bc3f-d}"
+    - "platform=linux"
+    - "permission_set=builder"
+    - "scaler_version=2"
+    - "environment=${CI_ENVIRONMENT:-cn-production}"
```
Comment (Helcaraxan) [inline .buildkite/premerge.steps.yaml:12]: Also please set `"node_stability=${CI_NODE_STABILITY:-interruptible}"` so we benefit of ~60% cost reduction.

File: .buildkite/premerge.steps.yaml (hunk 1 above)
Comment (stuarthu) [reply .buildkite/premerge.steps.yaml:7]: done

File: .buildkite/premerge.steps.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+common: &common
+  agents:
+    - "agent_count=1"
+    - "boot_disk_size_gb=400"
```
Comment (Helcaraxan) [inline .buildkite/premerge.steps.yaml:5]: This label should not be necessary either.

File: .buildkite/premerge.steps.yaml (hunk 2 above)
Comment (stuarthu) [reply .buildkite/premerge.steps.yaml:12]: done

File: .buildkite/premerge.steps.yaml (hunk 3 above)
Comment (stuarthu) [reply .buildkite/premerge.steps.yaml:5]: removed

**Pull request merged**